core/org.wso2.carbon.user.core/src/test/java/org/wso2/carbon/user/core/util/DatabaseUtilTest.java

```diff
@@ -0,0 +1,260 @@
+        log.info("Unhandled SQLRecoverableException");
+        exceptionNotHandled = true;
+        } catch (SQLException e) {
+        // this shouldn't happen
+        Assert.fail();
+        } finally {
+            Assert.assertFalse(exceptionNotHandled);
+        }
+    }
+
+    @Test
+    public void createUserStoreDataSource() throws Exception {
+
+        when(realmConfig.isLogAbandoned()).thenReturn(true);
+        when(realmConfig.isRemoveAbandoned()).thenReturn(true);
+        when(realmConfig.getRemoveAbandonedTimeout()).thenReturn(100);
+        DatabaseUtil.createUserStoreDataSource(realmConfig);
+
+        Assert.assertTrue(realmConfig.isLogAbandoned());
+        Assert.assertTrue(realmConfig.isRemoveAbandoned());
+        Assert.assertEquals(100, realmConfig.getRemoveAbandonedTimeout());
+        Assert.assertNotNull(datasource.getConnection());
+    }
+
+    @Test
+    public void nullCreateUserStoreDataSource() throws Exception {
+
+        DatabaseUtil.createUserStoreDataSource(realmConfig);
+
+        Assert.assertFalse(realmConfig.isLogAbandoned());
+        Assert.assertFalse(realmConfig.isRemoveAbandoned());
+        Assert.assertEquals(0, realmConfig.getRemoveAbandonedTimeout());
+
+        Assert.assertEquals(realmConfig.isLogAbandoned(), poolProperties.isLogAbandoned());
+        Assert.assertEquals(realmConfig.isRemoveAbandoned(), poolProperties.isRemoveAbandoned());
+        Assert.assertEquals(realmConfig.getRemoveAbandonedTimeout(), poolProperties.getRemoveAbandonedTimeout());
+    }
+
+    @Test
+    public void getStringValuesFromDatabaseShouldReturnValues() throws Exception{
+
+        ResultSetMocker resultSetMocker = new ResultSetMocker().invoke();
+        String sqlStmt = resultSetMocker.getSqlStmt();
+        List<Object> params = resultSetMocker.getParams();
+        List<Object> resultSetAnswers = resultSetMocker.getResultSetAnswers();
+
+        String[] results =   DatabaseUtil.getStringValuesFromDatabase(conn, sqlStmt, params.get(0), params.get(1), params.get(2));
+
+        Assert.assertEquals( resultSetAnswers.get(0), results[0]);
+        Assert.assertEquals(resultSetAnswers.get(1), results[1]);
+        Assert.assertEquals(resultSetAnswers.get(2), results[2]);
+    }
+
+
+    @Test (expected = UserStoreException.class)
+    public void getStringValuesFromDatabaseShouldHandleSQLRecoverableException() throws Exception{
+
+        // cannot handle sqlerecoverable for resultsets
+        ResultSetMocker resultSetMocker = new ResultSetMocker().invoke();
+        String sqlStmt = resultSetMocker.getSqlStmt();
+        List<Object> params = resultSetMocker.getParams();
+        Mockito.doThrow(new SQLRecoverableException("throw SQLRecoverableException")).when(resultSetMocker.get()).next();
+
+        String[] results =   DatabaseUtil.getStringValuesFromDatabase(conn, sqlStmt, params.get(0), params.get(1), params.get(2));
+        Assert.fail("Can not handle SQLRecoverableError" );

```

Comment:
@imesh Not really a typo. It was intended for emphasis.  But, you didn't like it, so I changed it..